planning/behavior_velocity_dynamic_obstacle_stop_module/src/scene_dynamic_obstacle_stop.cpp

```diff
@@ -93,98 +137,67 @@ bool DynamicObstacleStopModule::modifyPathVelocity(PathWithLaneId * path, StopRe
     make_forward_footprints(dynamic_obstacles, params_, hysteresis);
   const auto footprints_duration_us = stopwatch.toc("footprints");
   stopwatch.tic("collisions");
-  const auto collision =
-    find_earliest_collision(ego_data, dynamic_obstacles, obstacle_forward_footprints, debug_data_);
+  auto collisions = find_collisions(ego_data, dynamic_obstacles, obstacle_forward_footprints);
+  update_object_map(object_map_, collisions, clock_->now(), ego_data.path.points, params_);
```

Comment:
Not that important, but is it OK to call `update_object_map` before `find_collisions` and `hysteriysis` ?

Reply:
Yes it is okay. The object map is used to store the collision associated with each object and the times of first/last detection.
After this map is updated we can find the collision to avoid with `find_collisions`.

I will add some docstring to make the current logic more clear.

Reply:
I added docstrings [aa086cd](https://github.com/autowarefoundation/autoware.universe/pull/6683/commits/aa086cda15cf3c3aee4634ab54c5582cc0312205)